File: C#/PAT_B/1014.cs
using System;
class P_1014
{
    static string[] day = { "MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN" };
    static void Main()
    {
        // 输入4个变量
        var s1 = Console.ReadLine();
        var s2 = Console.ReadLine();
        var s3 = Console.ReadLine();
        var s4 = Console.ReadLine();
        var point = 0;
        for (var i = 0; i < s1!.Length; i += 1)
        {
            if (s1[i] == s2![i] && (s1[i] >= 'A' && s1[i] <= 'G'))
            {
                Console.Write("{0} ", day[s1[i] - 'A']);
                point = i;
                break;
            }
        }
        for (var i = point + 1; i < s1.Length; i += 1)
        {
            if (s1[i] == s2![i] && (s1[i] >= '0' && s1[i] <= '9'))
            {
                Console.Write("0{0}:", s1[i] - '0');
                break;
            }
            if (s1[i] == s2[i] && (s1[i] >= 'A' && s1[i] <= 'N'))
            {
                Console.Write("{0}:", s1[i] - 'A' + 10);
                break;
            }
        }
        var min = 0;
        for (var i = 0; i < s3!.Length; i += 1)
        {
            if ((s3[i] >= 'a' && s3[i] <= 'z') || (s3[i] >= 'A' && s3[i] <= 'Z'))
            {
                min += 1;
                if (s3[i] == s4![i])
                {
                    break;
                }
            }
        }
        if (min < 10)
        {
            Console.Write("0{0}", min);
        }
        else
        {
            Console.Write("{0}", min);
        }
    }
}

File: code/CPP/PAT(Basic_Level)Practice/1014.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;

const string day[] = { "MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN" };

int main() {
    string str1;
    string str2;
    string str3;
    string str4;
    cin >> str1;
    cin >> str2;
    cin >> str3;
    cin >> str4;
    auto first = 0;
    auto i = str1.length();
    for (i = 0; i < str1.length(); i++) {
        if (str1[i] == str2[i] && (str1[i] >= 'A' && str1[i] <= 'G')) {
            cout << day[str1[i] - 'A'] << ' ';
            first = i;
            break;
        }
    }

    for (i = first + 1; i < str1.length(); i++) {
        if (str1[i] == str2[i]) {
            if (str1[i] >= '0' && str1[i] <= '9') {
                cout << '0' << str1[i] - '0' << ':';
                break;
            }
            else if (str1[i] >= 'A' && str1[i] <= 'N') {
                cout << str1[i] - 'A' + 10 << ':';
                break;
            }
        }
    }

    for (i = 0; i < str3.length(); i++) {
        if (str3[i] == str4[i] && ((str3[i] >= 'A' && str3[i] <= 'Z') || (str3[i] >= 'a' && str3[i] <= 'z'))) {
            if (i < 10) {
                cout << '0' << i;
            }
            else {
                cout << i;
            }
            break;
        }
    }

    return 0;
}